[Ameba] Shell commands update

pankore opened the pull request:
- This PR provides all-clusters-app on Ameba platform a way to send commands as a client through the matter shell
- Implemented commands for Identify, OnOff, LevelControl, ColorControl, Thermostat clusters
- Allow matter shell to process input arguments for commands that require them
- Update all-clusters-app README

pankore committed:
[Shell] Read cluster attributes from matter shell
- Received data but DataResponse is still not printed out
- Support more OnOff commands for matter shell

pankore committed:
[Shell] enable inputting arguments for certain commands

pankore committed:
[Shell] Add levelcontrol commands to shell
- Separate different cluster commands into header files

pankore committed:
[Shell] Add colorcontrol commands to shell

pankore committed:
[Shell] Add thermostat commands to shell

pankore committed:
[Shell] Add handlers for group commands

pankore committed:
[README] Update readme for all-clusters-app

pankore committed:
[Shell] Add identify commands to shell

pankore committed:
[Shell] Increase the number of tokens to fit more arguments
- Add in missing colorcontrol commands
- Fix readme link

pankore committed:
[Shell] Enable reading of attribute list

pankore committed:
[Shell] Fix transition data type and argument errors
- Added logs for levelcontrol commands

pankore committed:
[Shell] Fix compile errors for when CONFIG_ENABLE_CHIP_SHELL is disabled
- Added CONFIG_ENABLE_CHIP_SHELL macro to block off code that requires shell

pankore committed:
[Resyle] fix styling

pankore commented:
Would like to request for this PR to be merged into 1.0-branch, thank you.

restyled-commits committed:
Restyled by prettier-markdown

pankore commented:
Spelling check is failing this PR wrongly. It is marking `currenty`, which is CurrentY, as a misspelled word of "currently"

pankore committed:
[Shell] Change ColorTemperature to ColorTemperatureMireds

pankore commented:
Hi @woody-apple, @andy31415, the misspell check is failling as It is marking `currenty`, (which refers to `CurrentY`), as a misspelled word of "currently". How should I fix this issue?

andy31415 commented on examples/all-clusters-app/ameba/README.md:
```diff
@@ -137,28 +137,70 @@ to be On or Off.
--   This example shows how to bind a Switch Device to a Lighting Device and
+-   To know what are the available subcommands and what it does, simply enter
+    the command that you are interested in
+
```
use backticks here since these are shell things. This avoids the spellchecker trying to think these are words (like `offWE` and such are not english words but rather code.

andy31415 commented on examples/all-clusters-app/ameba/main/BindingHandler.cpp:
```diff
@@ -354,27 +245,332 @@ static void RegisterSwitchCommands()
+        { &ColorControlReadCurrentSaturation, "currentsat", "Read currentsaturaion attribute" },
+        { &ColorControlReadRemainingTime, "remaintime", "Read remainingtime attribute" },
+        { &ColorControlReadCurrentX, "currentx", "Read currentx attribute" },
+        { &ColorControlReadCurrentY, "currenty", "Read currenty attribute" },
```
Currently here is misspelled

andy31415 replied:
I see, current X and Y are currents. We have to figure out how to format these. I am wondering if a help for `currentx` is helpful when we say `reads currentx attribute` as it seems to be kind of redundant. That does not seem to fix the fact that we need these non-words as commands. Need to check spellcheck for escape options.

pankore replied:
Changed `currentx` to `current-x` and `currenty` to `current-y` to avoid spell check failing

pankore committed:
[Readme] Update all-clusters-app readme

pankore committed:
[Shell] Change currentx and currenty to current-x and current-y respectively
- To avoid spell check failing as it identifies currenty as mispelling of currently

pankore committed:
[Readme] Restyle

pankore committed:
[Shell] Fix typecasting error

pankore committed:
[Shell] restyle
